lib/question.js

```diff
@@ -1011,14 +1012,57 @@ module.exports = {
 
                 const studentMessage = 'Error rendering question';
                 const courseData = {variant, question, submission, course};
-                this._writeCourseIssues(null, courseIssues, variant.id, locals.authn_user.user_id, studentMessage, courseData, (err) => {
+                // locals.authn_user may not be populated when rendering a panel
+                const user_id = (locals && locals.authn_user) ? locals.authn_user.user_id : null;
+                this._writeCourseIssues(null, courseIssues, variant.id, user_id, studentMessage, courseData, (err) => {
                     if (ERR(err, callback)) return;
                     return callback(null, htmls);
                 });
             });
         });
     },
 
+    _buildQuestionUrls(urlPrefix, variant, question, instance_question, assessment) {
```

Comment:
Function needs a comment block above it. I'm not sure these are really helping anything? I've been mandating them in `lib/question.js` and `lib/assessment.js` but we could just drop them if they aren't doing anything. Thoughts?

Reply:
In general I'm in favor of lots of documentation. I'll go ahead and add something.